joss-paper/paper.md

```diff
@@ -0,0 +1,202 @@
+the range and scale of physical models of interest continues to increase -- to
+reformulate the `21cmFAST` code in order to provide a fast, modular, well-documented,
+well-tested, stable simulator for the community.
+
+# Features of 21cmFAST v3
+
+This paper presents `21cmFAST` v3+, which enables these essential new features.
+While keeping the same core functionality of previous versions of `21cmFAST`, it has
+been fully integrated into a Python package, with a simple and intuitive interface, and
+a great deal more flexibility.
+At a higher level, in order to maintain best practices, a community of users and
+developers has coalesced into a formal collaboration which maintains the project via a
+Github organization.
+This allows the code to be consistently monitored for quality, maintaining high test
+coverage, stylistic integrity, dependable release strategies and versioning,
+and peer code review.
+It also provides a single point-of-reference for the community to obtain the code,
+report bugs and request new features (or get involved in development).
+
+A significant part of the work of moving to a Python interface has been the
+development of a robust series of underlying Python structures which handle the passing
+of data between Python and C via the `CFFI` library.
+This foundational work provides a platform for future versions to extend the scientific
+capabilities of the underlying simulation code.
+The primary *new* usability features of `21cmFAST` v3+ are:
+
+* Convenient (Python) data objects which simplify access to and processing of the various
+  fields that form the brightness temperature.
+* Enhancement of modularity: the underlying C functions for each step of the simulation
+  have been de-coupled, so that arbitrary functionality can be injected into the process.
+* Conversion of most global parameters to local structs to enable this modularity, and
+  also to obviate the requirement to re-compile in order to change parameters.
+* Simple `pip`-based installation.
+* Robust on-disk caching/writing of data, both for efficiency and simplified reading of
+  previously processed data (using HDF5).
+* Simple high-level API to generate either coeval cubes (purely spatial 3D fields defined
+  at a particular time) or full lightcone data (i.e. those coeval cubes interpolated over
+  cosmic time, mimicking actual observations).
+* Improved exception handling and debugging.
+* Convenient plotting routines.
+* Simple configuration management, and also more intuitive management for the
+  remaining C global variables.
+* Comprehensive API documentation and tutorials.
+* Comprehensive test suite (and continuous integration).
+* Strict `semantic versioning <https://semver.org>`_.
+
+While in v3 we have focused on the establishment of a stable and extendable infrastructure,
+we have also incorporated several new scientific features, appearing in separate papers:
+
```

Comment:
It would be good to add 
+ * Calculate the rest-frame UV luminosity functions based on the parametrized galaxy models.

I'm not sure it is good to put this here. But, the previous 21cmFAST didn't include this calculation, it was only in 21CMMC. Anyway, I think this feature should be pointed out somewhere.

Reply:
Thanks @jaehongpark00, I'll add that in.